plugins/module_utils/ds8000.py

```diff
@@ -0,0 +1,194 @@
+
+__metaclass__ = type
+
+import json
+import traceback
+import abc
+
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils import six
+
+REQUESTS_IMP_ERR = None
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+    HAS_REQUESTS = False
+
+PYDS8K_IMP_ERR = None
+try:
+    from pyds8k.client.ds8k.v1.client import Client
+    HAS_PYDS8K = True
+except ImportError:
+    PYDS8K_IMP_ERR = traceback.format_exc()
+    HAS_PYDS8K = False
+
+DEFAULT_BASE_URL = '/api/v1'
+DEFAULT_POOLS_URL = '/pools'
+PRESENT = 'present'
+ABSENT = 'absent'
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Ds8000ManagerBase(object):
+    def __init__(self, module):
+
+        if not HAS_REQUESTS:
+            module.fail_json(msg=missing_required_lib('requests'),
+                             exception=REQUESTS_IMP_ERR)
+
+        if not HAS_PYDS8K:
+            module.fail_json(msg=missing_required_lib('pyds8k'),
+                             exception=PYDS8K_IMP_ERR)
+
+        self.module = module
+        self.params = module.params
+        self.hostname = module.params['hostname']
+        self.username = module.params['username']
+        self.password = module.params['password']
+        self.client = self.connect_to_api()
+        self.token = self.get_rest_api_token()
+        self.headers = {'Content-Type': 'application/json'}
+        self.token_header = {'X-Auth-Token': '{token}'.format(token=self.token)}
+        self.headers.update(self.token_header)
+        self.changed = False
+        self.failed = False
+
+    def get_all_volumes(self):
+        volumes = []
+        pools = self.get_ds8000_objects_by_type(DEFAULT_POOLS_URL, 'pools')
+        for pool in pools:
+            volumes_by_pool = self.get_ds8000_objects_by_type(
+                '/pools/{pool}/volumes'.format(pool=pool['name']), 'volumes')
+            volumes.extend(volumes_by_pool)
+        return volumes
+
+    def verify_ds8000_object_exist(self, ds8000_object_param_name,
+                                   ds8000_object_type):
+        ds8000_object_name = self.params[ds8000_object_param_name]
+        if self.does_ds8000_object_exist(ds8000_object_param_name, ds8000_object_type):
+            return True
+        self.module.fail_json(
+            msg="The {object_type} {object_name} "
+                "does not exist in the ds8000 storage.".format(
+                    object_type=ds8000_object_type[:-1], object_name=ds8000_object_name))
+        return False
+
+    def does_ds8000_object_exist(self, ds8000_object_param_name,
+                                 ds8000_object_type):
+        ds8000_object_name = self.params[ds8000_object_param_name]
+        ds8000_objects = self.get_ds8000_objects_name_by_type(ds8000_object_type)
+        if ds8000_object_name not in ds8000_objects:
+            return False
+        return True
```

Comment:
We can simplify this by:
```suggestion
        return ds8000_object_name in ds8000_objects
```